### ISSUE TITLE
Tasks dont appear in Firefox

### ISSUE THREAD

**Issue by marissa999** [cdr/code-server#1914]
The tasks do not appear in Firefox, but they appear as expected in Chromium. Firefox just tells me that there are no configured tasks, meanwhile in Chromium I am able to see and execute the tasks.

- Web Browser: Firefox (doesn't work as expected)
- Local OS: Arch Linux
- Remote OS: Debian GNU/Linux 10 (buster) in a LXC container on Proxmox
- Remote Architecture: X86
- `code-server --version`: 3.4.1 48f7c2724827e526eeaa6c2c151c520f48a61259

**Comment by marissa999** [comment]
I just realised that the tasks properly appear in Firefox in Private Mode. Therefore I heavily assume that a setting in my browser is blocking something. I disabled all extensions in Firefox and still was not able to see the tasks appear in the "Run Task"-Menu in the Command Palette.

If anybody has a guess what setting in my browser might prevent the tasks from working properly please tell me.

**Comment by code-asher** [comment]
The only thing that comes immediately to mind is that I think the tasks don't work if you aren't browsing using a secure context (usually either https or localhost).

If that's not the case for you then is there anything in the browser console logs?

**Comment by marissa999** [comment]
I use nginx as a reverse proxy which takes care of https. And considering that it works in Google Chrome and Firefox in private mode I think it's safe to say that the first thing you mentioned can't be the problem, unless Firefox in non private mode has different security policies than Firefox in private mode or Google Chrome.

I checked the console logs and the network tab and couldn't see anything suspicious.

**Comment by code-asher** [comment]
It was definitely a long shot. :stuck_out_tongue: Maybe it's some kind of caching issue? I spun up 3.4.1 locally but wasn't able to replicate with FF 77.0.1.

**Comment by marissa999** [comment]
And because the issue is not weird enough already: 
In my workspace with Rust the tasks appear as they should (task-type is cargo).
In my folder with a C-Project the tasks do not appear (task-types are shell, and execute either make or cmake).

I will play around a bit with workspaces and task-types. Hopefully I can figure something out. 

Btw, I found somebody on reddit having the same problem: https://www.reddit.com/r/codeserver/comments/hkhe93/tasksjson_file_being_ignored/

**Comment by marissa999** [comment]
https://github.com/cdr/code-server/issues/1616 maybe related?

**Comment by smoothchocolate** [comment]
I am confirming that I have the same behavior with Firefox 79.0 on Linux in which tasks.json gets ignored, although when setting Firefox to private mode, tasks.json is parsed correctly. Tasks.json is also parsed correctly on Google Chrome 84.0.4147.105.

My backend is run with NGINX as a reverse proxy.

**Comment by alkenes** [comment]
I can also confirm that Firefox ignores the tasks file unless in private browsing.

**Comment by code-asher** [comment]
I recently ran into this on Firefox finally, and then on Chrome actually. So I was able to debug and it looks like there are two task providers, one that runs in the Node extension host and one that runs in the web worker extension host. The latter does not support tasks while the former does. So depending on the order they load they will either enable or disable tasks.

I'm not sure what the ideal solution is but I think it might be to only let the Node side register and don't allow the web worker to override it, at least for now.
